test/e2e/es/certs_test.go

```diff
@@ -32,24 +32,133 @@ import (
 )
 
-func TestCustomTransportCA(t *testing.T) {
-	caSecretName := "my-custom-ca"
-	esName := "test-custom-ca"
+var caSecretName = "my-custom-ca"
+
+func TestCustomHTTPCA(t *testing.T) {
+	esName := "test-custom-http-ca"
 	esNamespace := test.Ctx().ManagedNamespace(0)
 
-	mkTestSecret := func(cert, key []byte) corev1.Secret {
-		return corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: esNamespace,
-				Name:      caSecretName,
-			},
-			Data: map[string][]byte{
-				certificates.CertFileName: cert,
-				certificates.KeyFileName:  key,
-			},
-		}
+	// Create a multi-node cluster so we have transient states when switching certs where some nodes still have the old ones
+	initialCluster := elasticsearch.NewBuilder(esName).
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources).
+		// transient error during cert changes, this affects also this builder as we migrate back to built-in certs at the end
+		WithToleratedHealthCheckErrors("certificate signed by unknown authority")
+
+		// start with the operator provided HTTP CA
+	withBuiltinCA := test.WrappedBuilder{
+		BuildingThis: initialCluster,
+		PreDeletionSteps: func(k *test.K8sClient) test.StepList {
+			return test.StepList{
+				{
+					Name: "Clean up custom CA secret",
+					Test: func(t *testing.T) {
+						// This is counter-intuitive but deletion steps run on the initial builder
+						// let's clean up the CA secret here.
+						toDelete := mkCertSecret(nil, nil)
+						_ = k.Client.Delete(context.Background(), &toDelete)
+					},
+				},
+			}
+		},
+	}
+
+	// The initial Cluster builder should result in a healthy cluster as verified by the standard check steps
+	// Now modify cluster to use custom HTTP CA but simulate a user error by populating the secret with
+	// garbage and verify this is bubbled up through an event
+	b := initialCluster.
+		WithCustomHTTPCerts(caSecretName).
+		WithMutatedFrom(&initialCluster)
+
+	withBogusCert := test.WrappedBuilder{
+		BuildingThis: b,
+
+		// we don't want to run a regular mutation as the invalid certificates will not lead to a successful upgrade
+		OverrideMutationSteps: func(k *test.K8sClient) test.StepList {
+			steps := test.StepList{
+				{
+					Name: "Create an invalid CA secret",
+					Test: test.Eventually(func() error {
+						bogusSecret := mkCertSecret([]byte("garbage"), []byte("more garbage"))
+						_, err := reconciler.ReconcileSecret(k.Client, bogusSecret, nil)
+						return err
+					}),
+				},
+			}
+			steps = append(steps, b.UpgradeTestSteps(k)...)
+			steps = append(steps,
+				test.Step{
+					Name: "Invalid CA secret should create events",
+					Test: test.Eventually(func() error {
+						eventList, err := k.GetEvents(test.EventListOptions(esNamespace, initialCluster.Elasticsearch.Name)...)
+						if err != nil {
+							return err
+						}
+						for _, evt := range eventList {
+							if evt.Type == corev1.EventTypeWarning &&
+								evt.Reason == events.EventReconciliationError &&
```

Comment:
I would include a namespaced name comparison to check that the event has not been generated by a previous run (it happened on my dev environment), or a similar test that we may add in the future.
```suggestion
								evt.Reason == events.EventReconciliationError &&
								evt.InvolvedObject.Namespace == b.Elasticsearch.Namespace &&
								evt.InvolvedObject.Name == b.Elasticsearch.Name &&
```